runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -464,6 +467,36 @@ def teardown(self):
         Example:
             >>> rh.ondemand_cluster("rh-cpu").teardown()
         """
+        # TODO [SB]: remove the den_auth check once we will get status of clusters without den_auth as well.
+        warning_msg = "Failed to update in Den that the cluster is terminated."
+        if self.den_auth:
+            try:
+                cluster_status_data = self.status()
+                status_data = {
+                    "status": ResourceServerStatus.terminated,
+                    "resource_type": self.__class__.__base__.__name__.lower(),
```

Comment:
i think we can do: `self.__class__.__name__.lower()`? Also i think we use this elsewhere to track the `resource_subtype`, might be worth creating a property for that